scripts/ASTImportTest.sh

```diff
@@ -131,6 +146,125 @@ function test_ast_import_export_equivalence
+            return 1
+        fi
+    fi
+
+    for input_json in "expected.json" "expected.optimize.json"
+    do
+        local optimize_param=""
+        if [[ "$input_json" == "expected.optimize.json" ]]
+        then
+            optimize_param="--optimize"
+        fi
+
+        # Note that we have some test files, that only consists of free functions.
+        # Those files doesn't define any contracts, so the resulting json does not define any
+        # keys. In this case `jq` returns an error like `jq: error: null (null) has no keys`
+        # to not get spammed by these errors, errors are redirected to /dev/null.
+        for contract in $(jq '.contracts | keys | .[]' $input_json 2> /dev/null)
+        do
+            for output in "${outputs[@]}"
+            do
+                jq --raw-output ".contracts.${contract}.\"${output}\"" "$input_json" > "expected.${output}.json"
+            done
+
+            assembly=$(cat expected.asm.json)
+            [[ $assembly != "" && $assembly != "null" ]] || continue
+
+            local import_command=("${SOLC}" --combined-json "bin,bin-runtime,opcodes,asm,srcmap,srcmap-runtime" --pretty-json --json-indent 4 --import-asm-json expected.asm.json)
+            if ! "${import_command[@]}" > obtained.json 2> obtained.error
+            then
+                success=0
+                printError "ERROR: (import) EVM Assembly JSON reimport failed for ${sol_file}"
+                if [[ $EXIT_ON_ERROR == 1 ]]
+                then
+                    print_used_commands "$(pwd)" "${export_command[*]} ${optimize_param}" "${import_command[*]}"
+                    return 1
+                fi
+            fi
+
+            for output in "${outputs[@]}"
+            do
+                for obtained_contract in $(jq '.contracts | keys | .[]' obtained.json  2> /dev/null)
+                do
+                    jq --raw-output ".contracts.${obtained_contract}.\"${output}\"" obtained.json > "obtained.${output}.json"
+                    # compare expected and obtained evm assembly json
+                    if ! diff_files "expected.${output}.json" "obtained.${output}.json"
+                    then
+                        success=0
+                        printError "ERROR: (${output}) EVM Assembly JSON reimport failed for ${sol_file}"
+                        if [[ $EXIT_ON_ERROR == 1 ]]
+                        then
+                            print_used_commands "$(pwd)" "${export_command[*]} ${optimize_param}" "${import_command[*]}"
+                            return 1
+                        fi
+                    fi
+                done
+            done
+
+            # direct export via --asm-json, if imported with --import-asm-json.
+            if ! "${SOLC}" --asm-json --import-asm-json expected.asm.json --pretty-json --json-indent 4 | tail -n+2 > obtained_direct_import_export.json 2> obtained_direct_import_export.error
```

Comment:
Not sure if it's really useful to have this. I mean, if possible, we should replace `--combined-json` with I think we cannot do this without quite a bit of extra work (would require using Standard JSON or parsing the CLI outputs). If we can't get rid of it completely, we can just as well keep using `--combined-json` for the time being.

Or is this meant to do what @ekpyron suggested in https://github.com/ethereum/solidity/pull/13673#discussion_r1016492698? But in that case there's no meaningful difference between this and the result of `$import_command` above. Both do the same reimport and just obtain the output via different interface. Comparing `expected.bin.json` with `obtained.bin.json` is basically what basically @ekpyron wanted. The variant with optimization would be `expected.optimize.bin.json` with `obtained.optimize.bin.json` (which I think you don't have).

Reply:
I was also not sure whether this is really useful in practice, e.g. will a user ever need this. But I thought that this is a nice test. 

With regards to the optimisation, during a call with @ekpyron we decided that the evm json import should not allow running the optimiser again. I should have write it down somewhere in this PR. This simplified the import code a bit, where on the other hand users will probably not need to run the optimiser again on specific code, because the user already may have already run the optimiser during export, or the user already optimised the byte code with an external tool before import.